src/plugins/data/common/datasets/types.ts

```diff
@@ -28,6 +28,8 @@ export interface DataSourceMeta {
   name?: string;
   /** Optional session ID for faster responses when utilizing async query sources */
   sessionId?: string;
+  /** Optional requiresTimeFilter determines if a time filter is needed */
+  requiresTimeFilter?: boolean;
```

Comment:
`supportsTimeFilter` might be a better name here, since it's possible that we choose no time field even if there are time fields available. One of the options in the time field drop down (if there are time fields) is `I don't want to use the time filter`.

Reply:
`supportsTimeFilter` can change the name, if you have a hard opinion. But I disagree with your point that SQL and PPL async queries don't support time filter. We might have opted out of time filters support for these due to time constraints or product requirements. 

Example of time filters in Async SQL/PPL:
```
SELECT * FROM sample_table  WHERE  startTime < CAST('01-sep-2022' as date) AND (startTime >= CAST('02-sep-2022' as date) 

source = sample_table | where startTime < '09-01-2022:00.00.00.000' and startTime >= '09-02-2022:00.00.00.000' 
```